src/bun.js/bindings/Path.cpp

```diff
@@ -43,17 +42,10 @@ using namespace JSC;
      } \
     auto clientData = WebCore::clientData(vm); \
     auto isWindows = thisObject->get(globalObject, clientData->builtinNames().isWindowsPrivateName()); \
-    JSC::JSValue result = JSC::JSValue::decode( \
-        ZigFunction(globalObject, isWindows.asBoolean(), reinterpret_cast<JSC__JSValue*>(arguments.data()), argCount) \
-    ); \
-    JSC::JSObject *obj = result.getObject(); \
-    if (UNLIKELY(obj != nullptr && obj->isErrorInstance())) { \
-        scope.throwException(globalObject, obj); \
-        return JSC::JSValue::encode(JSC::jsUndefined()); \
-    } \
+    auto result = ZigFunction(globalObject, isWindows.asBoolean(), reinterpret_cast<JSC__JSValue*>(arguments.data()), argCount); \
     if (UNLIKELY(scope.exception())) \
         return JSC::JSValue::encode(JSC::jsUndefined()); \
-    return JSC::JSValue::encode(result);
+    return result;
```

Comment:
☝️ Tracked down why `return .zero` from path methods would cause a segfault. Now it's safe to `return .zero`.

Reply:
thanks for removing "if this function returns an error, throw it", this pattern is horrible and should never be used. we should always throw errors.

the segfault was a null pointer dereference on the removed `.getObject()`. this will assert in debug mode that the cell is non-zero (0 is classified as a cell type, aka a pointer)

i think we could get away with also removing `if (UNLIKELY(scope.exception()))` too, since if there is an exception thrown, the jsvalue return is ignored, which is where we use zero. crashes due to following this rule are indications of other mistakes in our code.